Adding fuzz directory and modifications to support it.

catenacyber opened the pull request:
Adds a new target to fuzz the mysql client

Quickly finds 2 NULL derefs, but looks ok after adding checks for these

catenacyber commented:
I confirm the code being submitted is offered under the terms of the OCA, and that I am authorized to contribute it.

mysql-admin commented:
Hi @catenacyber @ChAlv ,
Thanks for contributing to MySQL
As this contribution was authored by @ChAlv , @ChAlv is the one that is authorized to contribute it and the person that needs to confirm it is submitted under the terms of the (his) OCA. 

@ChAlv Please confirm this code is submitted under the terms of the OCA (Oracle's Contribution Agreement) you have previously signed by cutting and pasting the following text as a comment:
"I confirm the code being submitted is offered under the terms of the OCA, and that I am authorized to contribute it."

Thanks
==Omer

catenacyber commented:
Sorry if I messed up my git work.

This was a work started by Christophe Alves, me and 2 other (in https://github.com/google/oss-fuzz/pull/2593 ), but I changed almost all of it, re-squashing everything into the first commit.

Should I open a new PR with the commit as my own ?

mysql-admin commented:
Hi @catenacyber ,
I different PR is probably the simplest way to address this as we need to have a match between the author and the OCA submitter 

Thanks and apologies for the 'detour' 
==Omer

catenacyber committed:
client: add fuzz target

Defines a new vio which simply uses a buffer in memory, so that
the fuzz target initializes it with the fuzz input, and it gets
used as if it were the data from the network sent by the server.

catenacyber commented:
@mysql-admin is it better this way ?
